test_communication/test/subscriber_py.py

```diff
@@ -62,7 +62,7 @@ def listener(message_name, namespace):
     spin_count = 1
     print('subscriber: beginning loop')
     while (rclpy.ok() and len(received_messages) != len(expected_msgs)):
-        rclpy.spin_once(node)
+        rclpy.spin_once(node, timeout_sec=0)
```

Comment:
I am not sure what the semantic of zero is here, but I would expect it to either return immediately or block forever which both seems not desirable.

Reply:
The default `None` is block forever. 0 is execute immediately available work and return otherwise.

Reply:
Makes sense since the new argument changes behavior. That means this change makes this a busy loop which isn't acceptable which will consume all of a CPU core.

I suggest to pass a timeout of e.g. `0.1s` instead.

Reply:
Good point

Reply:
Comparing this to `replier_py.py` maybe a larger timeout makes sense - to match the other test and avoid printing too frequently.